### PR TITLE
(FB: 157140) - Show missing parent errors

### PR DESCRIPTION
**Pull request by benrudolph**
Small refactor of error handling in bulk importer

http://manage.dimagi.com/default.asp?157140
screen shot 2015-02-16 at 2 00 20 pm

@twymer

### CODE REVIEW

File: corehq/apps/importer/const.py [hunk 1]
```diff
@@ -1,2 +1,10 @@
 class LookupErrors:
     NotFound, MultipleResults = range(2)
+
+class ImportErrors:
```
Comment (dimagimon) [inline corehq/apps/importer/const.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/importer/util.py [hunk 2]
```diff
@@ -204,6 +206,51 @@ class InvalidDateException(Exception):
...
+    def __init__(self, *args, **kwargs):
+        self.errors = defaultdict(dict)
+
+    def add(self, error, row_number):
+        self.errors[error]['error'] = _(error)
+
+        try:
+            self.errors[error]['description'] = self.ERROR_MSG[error]
+        except KeyError:
+            self.errors[error]['description'] = ERROR_MSG[ImportErrors.CaseGeneration]
```
Comment (dimagimon) [inline corehq/apps/importer/util.py]: F821 undefined name 'ERROR_MSG'

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/importer/tasks.py [hunk 3]
```diff
@@ -237,10 +235,6 @@ def _submit_caseblocks(caseblocks):
         'created_count': created_count,
         'match_count': match_count,
         'too_many_matches': too_many_matches,
-        'blank_externals': blank_external_ids,
-        'invalid_dates': invalid_dates,
-        'owner_id_errors': owner_id_errors,
-        'owner_name_errors': owner_name_errors,
-        'errors': errors,
+        'errors': errors.as_dict(),
```
Comment (twymer) [inline corehq/apps/importer/tasks.py]: Aw hell yes.

:tada::balloon::fireworks:

**Comment by twymer** [comment]
I am SO into this PR.

File: corehq/apps/importer/const.py [hunk 4]
```diff
@@ -1,2 +1,11 @@
...
     NotFound, MultipleResults = range(2)
+
+
+class ImportErrors:
+    InvalidOwnerName = 'Invalid Owner Name'
+    InvalidOwnerId = 'Invalid Owner ID'
+    InvalidParentId = 'Invalid Parent ID'
+    InvalidDate = 'Invalid Date'
+    BlankExternalId = 'Blank External ID'
+    CaseGeneration = 'Case Generation Error'
```
Comment (millerdev) [inline corehq/apps/importer/const.py]: Should these be translated?

Comment (benrudolph) [inline corehq/apps/importer/const.py]: It's translated here: https://github.com/dimagi/commcare-hq/pull/5554/files#diff-aade6bc0f6a5a9e3de32a4322a87fafdR238

I thought it'd be weird if you were debugging on a different locale and the errors were in a different language. Not exactly sure how to best deal with this; do you know best practice?

Comment (millerdev) [inline corehq/apps/importer/const.py]: > It's translated here: ...

I was looking for that, but couldn't find it. Will the translation engine know how to find these strings when they're collected and sent to the translators? My understanding is that a kind of static analysis is done on the code to find strings to translate, and it seems unlikely that these strings would be found by that system. @twymer can you verify that?

Comment (czue) [inline corehq/apps/importer/const.py]: @millerdev is right - you should use either `ugettext_lazy` or `ugettext_noop` here to mark them for static analysis.

Comment (benrudolph) [inline corehq/apps/importer/const.py]: ok i'll PR the change

Comment (benrudolph) [inline corehq/apps/importer/const.py]: https://github.com/dimagi/commcare-hq/pull/5584

Comment (czue) [inline corehq/apps/importer/const.py]: fyi: http://commcare-hq.readthedocs.org/en/latest/translations.html

**Pull request merged**